presto-hive/src/main/java/com/facebook/presto/hive/HiveUtil.java

```diff
@@ -1036,9 +1036,15 @@ else if (isLongDecimal(type)) {
             Integer physicalOrdinal = physicalNameOrdinalMap.get(column.getName());
             if (physicalOrdinal == null) {
                 // if the column is missing from the file, assign it a column number larger
-                // than the number of columns in the file so the reader will fill it with nulls
-                physicalOrdinal = nextMissingColumnIndex;
-                nextMissingColumnIndex++;
+                // than the number of columns in the file so the reader will fill it with nulls.
+                // If the index is negative, e.g. partitioning columns, leave it as is.
```

Comment:
`e.g. partitioning columns` is not quite accurate; this could be $path or $bucket synthesized column; perhaps, update to `e.g. partitioning columns, $bucket or $path`